packages/lucia/src/auth/request.ts

```diff
@@ -1,228 +1,83 @@
-		const sessionId = session?.sessionId ?? null;
-		if (this.storedSessionId === sessionId) return;
-		this.storedSessionId = sessionId;
-		this.requestContext.setCookie(this.auth.createSessionCookie(session));
-		if (session) {
-			debug.request.notice("Session cookie stored", session.sessionId);
-		} else {
-			debug.request.notice("Session cookie deleted");
-		}
-	};
+	public deleteSessionCookie(): void {
+		if (this.sessionCookie === null) return;
+		this.sessionCookie = null;
+		this.validatePromise = null;
+		this.setCookie(this.auth.createBlankSessionCookie());
+	}
 
-	public validate = async (): Promise<Session | null> => {
-		if (this.validatePromise) {
-			debug.request.info("Using cached result for session validation");
-			return this.validatePromise;
-		}
-		this.validatePromise = new Promise(async (resolve, reject) => {
-			if (!this.storedSessionId) return resolve(null);
-			try {
-				const session = await this.auth.validateSession(this.storedSessionId);
-				if (session.fresh) {
-					this.maybeSetSession(session);
+	public async validate(): Promise<
+		{ user: User; session: Session } | { user: null; session: null }
+	> {
+		if (!this.validatePromise) {
+			this.validatePromise = new Promise(async (resolve) => {
+				if (!this.sessionCookie) {
+					return resolve({ session: null, user: null });
 				}
-				return resolve(session);
-			} catch (e) {
-				if (
-					e instanceof LuciaError &&
-					e.message === "AUTH_INVALID_SESSION_ID"
-				) {
-					this.maybeSetSession(null);
-					return resolve(null);
+				const result = await this.auth.validateSession(this.sessionCookie);
+				if (result.session && result.session.fresh) {
+					const sessionCookie = this.auth.createSessionCookie(
+						result.session.id
+					);
+					this.setCookie(sessionCookie);
 				}
-				return reject(e);
-			}
-		});
-
+				return resolve(result);
+			});
+		}
 		return await this.validatePromise;
-	};
+	}
 
-	public validateBearerToken = async (): Promise<Session | null> => {
-		if (this.validateBearerTokenPromise) {
-			debug.request.info("Using cached result for bearer token validation");
-			return this.validatePromise;
-		}
-		this.validatePromise = new Promise(async (resolve, reject) => {
-			if (!this.bearerToken) return resolve(null);
-			try {
-				const session = await this.auth.validateSession(this.bearerToken);
-				return resolve(session);
-			} catch (e) {
-				if (e instanceof LuciaError) {
-					return resolve(null);
+	public async validateBearerToken(): Promise<
+		{ user: User; session: Session } | { user: null; session: null }
+	> {
+		if (!this.validateBearerTokenPromise) {
+			this.validateBearerTokenPromise = new Promise(async (resolve, reject) => {
+				if (!this.bearerToken) {
+					return resolve({ session: null, user: null });
 				}
-				return reject(e);
-			}
-		});
-
-		return await this.validatePromise;
-	};
+				return await this.auth.validateSession(this.bearerToken);
```

Comment:
```suggestion
				const result = await this.auth.validateSession(this.bearerToken);
				return resolve(result)
```

Currently this hangs when attempting to call validateBearerToken